File: Data Structures & Algorithms/validate-parentheses/submission-11.py
class Solution:
    def isValid(self, s: str) -> bool:
        ans = []
        mapping1 = {'[':']','(':')','{':'}'}
        # mapping2 = {']':'[',')':'(','}':'{'}
        for c in s:
            if c in mapping1:
                ans.append(mapping1[c])
            if c in ']})':
                if len(ans) == 0:
                    return False
                if ans[-1] != c:
                    return False
                ans.pop()

        if len(ans) != 0:
            return False

        return True

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-11.js
class Solution {
    /**
     * @param {string[]} tokens
     * @return {number}
     */
    evalRPN(tokens) {
        const ans = []
        tokens.forEach(i => {
            // console.log({i, ans})
            if ('+-/*'.includes(i)) {
                let a = ans.pop()
                let b = ans.pop()
                if (i === '+') {
                    // console.log(`${a} + ${b}`)
                    ans.push(a + b);
                } else if (i === '-') {
                    // console.log(`${a} - ${b}`)
                    ans.push(b - a);
                } else if (i === '/') {
                    // console.log(`${a} / ${b}`)
                    ans.push(Math.trunc(b / a));
                } else {
                    // console.log(`${a} * ${b}`)
                    ans.push(a * b);
                }
            } else {
                ans.push(Number(i))
            }
        });
        return ans[0]
    }
}
